components/create_post/create_post.jsx

```diff
@@ -638,16 +629,15 @@ export default class CreatePost extends React.Component {
         const ctrlOrMetaKeyPressed = e.ctrlKey || e.metaKey;
         const messageIsEmpty = this.state.message.length === 0;
         const draftMessageIsEmpty = this.props.draft.message.length === 0;
-        const ctrlEnterKeyCombo = this.props.ctrlSend && Utils.isKeyPressed(e, KeyCodes.ENTER) && ctrlOrMetaKeyPressed;
+        const ctrlEnterKeyCombo = Utils.isKeyPressed(e, KeyCodes.ENTER) &&
+            ((this.props.ctrlSend && ctrlOrMetaKeyPressed) || (!this.props.ctrlSend && !ctrlOrMetaKeyPressed)) &&
+            !e.altKey && !e.shiftKey;
```

Comment:
That's quite the one-liner, maybe split it out into a function that's a few lines so it's easier to read?

Reply:
@hmhealey I'm going to leave that to you since you wrote it and I struggle to understand it.

Reply:
@mkraft We could do
```javascript
const enterPressed = Utils.isKeyPressed(e, KeyCodes.ENTER) && !e.altKey && !e.shiftKey;
const ctrlPressedIfNecessary = (this.props.ctrlSend && ctrlOrMetaKeyPressed) || (!this.props.ctrlSend && !ctrlOrMetaKeyPressed);
const ctrlEnterKeyCombo = enterPressed && ctrlPressedIfNecessary;
```